streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java

```diff
@@ -2912,34 +2872,34 @@ void runOnce() {
     }
 
     private TaskManager mockTaskManagerPurge(final int numberOfPurges) {
-        final TaskManager taskManager = EasyMock.createNiceMock(TaskManager.class);
+        final TaskManager taskManager = mock(TaskManager.class);
         final Task runningTask = mock(Task.class);
         final TaskId taskId = new TaskId(0, 0);
 
-        expect(runningTask.state()).andReturn(Task.State.RUNNING).anyTimes();
-        expect(runningTask.id()).andReturn(taskId).anyTimes();
-        expect(taskManager.allTasks())
-                .andReturn(Collections.singletonMap(taskId, runningTask)).anyTimes();
-        expect(taskManager.commit(Collections.singleton(runningTask))).andReturn(1).anyTimes();
-        taskManager.maybePurgeCommittedRecords();
-        EasyMock.expectLastCall().times(numberOfPurges);
-        EasyMock.replay(taskManager, runningTask);
+        when(runningTask.state()).thenReturn(Task.State.RUNNING);
+        when(runningTask.id()).thenReturn(taskId);
+        when(taskManager.allTasks())
+            .thenReturn(Collections.singletonMap(taskId, runningTask));
+        when(taskManager.commit(Collections.singleton(runningTask))).thenReturn(1);
+        for (int i = 0; i < numberOfPurges; i++) {
```

Comment:
This for loop was added to actually simulate the number of purges that I assume the `mockTaskManagerPurge` is trying to emulate. Not sure how `EasyMock` did this but this version is more explicit and clear.